applications/MedApplication/tests/test_med_model_part_io.py

```diff
@@ -110,13 +101,174 @@ def test_quadrilateral_4N_linear_mesh(self):
         raise NotImplementedError
 
     def test_tetrahedra_4N_linear_mesh(self):
-        self.skipTest("This test is not yet implemented")
-        raise NotImplementedError
+        def mp_check_fct(model_part):
+            self.assertEqual(model_part.NumberOfNodes(), 36)
+            self.assertEqual(model_part.NumberOfGeometries(), 167)
+
+            # check how many geoms of each type
+            exp_geoms = {KM.Tetrahedra3D4: 65, KM.Triangle3D3: 64, KM.Line3D2: 32, KM.Geometry: 6}
+            self.assertEqual(sum(exp_geoms.values()), model_part.NumberOfGeometries())
+            self.assertDictEqual(exp_geoms, get_num_geometries_by_type(model_part))
+
+            self.assertAlmostEqual(KratosMed.MedTestingUtilities.ComputeLength(model_part), 3200)
+            self.assertAlmostEqual(KratosMed.MedTestingUtilities.ComputeArea(model_part), 340000)
+            self.assertAlmostEqual(KratosMed.MedTestingUtilities.ComputeVolume(model_part), 10000000)
+            self.assertAlmostEqual(KratosMed.MedTestingUtilities.ComputeDomainSize(model_part), 10343200)
+
+            for node in model_part.Nodes:
+                self.assertTrue(0.0 <= node.X <= 500.0)
+                self.assertTrue(0.0 <= node.X0 <= 500.0)
+
+                self.assertTrue(0.0 <= node.Y <= 100.0)
+                self.assertTrue(0.0 <= node.Y0 <= 100.0)
```

Comment:
oh no python, why do chained comparison operators work like this? :'(

Reply:
I like it :sweat_smile: